src/main/java/org/datadog/jenkins/plugins/datadog/DatadogHttpRequests.java

```diff
@@ -1,131 +1,168 @@
-      } else {
-        logger.severe(String.format("Client error: %s", e.toString()));
-      }
-      return false;
-    } finally {
-      if (conn != null) {
-        conn.disconnect();
-      }
-      return true;
+
+    /**
+     * Posts a given {@link JSONObject} payload to the Datadog API, using the
+     * user configured apiKey.
+     *
+     * @param payload - A JSONObject containing a specific subset of a builds metadata.
+     * @param type    - A String containing the URL subpath pertaining to the type of API post required.
+     * @return a boolean to signify the success or failure of the HTTP POST request.
+     * @throws IOException if HttpURLConnection fails to open connection
+     */
+    public static Boolean post(final JSONObject payload, final String type) throws IOException {
+        String urlParameters = "?api_key=" + Secret.toString(DatadogUtilities.getApiKey());
+        HttpURLConnection conn = null;
+        boolean status = false;
+
+        try {
+            logger.finer("Setting up HttpURLConnection...");
+            conn = DatadogHttpRequests.getHttpURLConnection(new URL(DatadogUtilities.getTargetMetricURL() + type + urlParameters));
+            conn.setRequestMethod("POST");
+            conn.setRequestProperty("Content-Type", "application/json");
+            conn.setUseCaches(false);
+            conn.setDoInput(true);
+            conn.setDoOutput(true);
+            OutputStreamWriter wr = new OutputStreamWriter(conn.getOutputStream(), "utf-8");
+            logger.finer("Writing to OutputStreamWriter...");
+            wr.write(payload.toString());
+            wr.close();
+            BufferedReader rd = new BufferedReader(new InputStreamReader(conn.getInputStream(), "utf-8"));
+            StringBuilder result = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                result.append(line);
+            }
+            rd.close();
+            JSONObject json = (JSONObject) JSONSerializer.toJSON(result.toString());
+            if ("ok".equals(json.getString("status"))) {
+                logger.finer(String.format("API call of type '%s' was sent successfully!", type));
+                logger.finer(String.format("Payload: %s", payload));
+                return true;
+            } else {
+                logger.fine(String.format("API call of type '%s' failed!", type));
+                logger.fine(String.format("Payload: %s", payload));
+                return false;
+            }
+        } catch (Exception e) {
+            if (conn.getResponseCode() == DatadogBuildListener.HTTP_FORBIDDEN) {
+                logger.severe("Hmmm, your API key may be invalid. We received a 403 error.");
+            } else {
+                logger.severe(String.format("Client error: %s", e.toString()));
+            }
+        } finally {
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+        return status;
```

Comment:
Should status be modified instead of returning on line 148? https://github.com/DataDog/jenkins-datadog-plugin/pull/160/files#diff-159b3bfc2922dfee4cd88de7f48a356bR148